### PR TITLE
[RA2 Ch6]: Clarifications to container image handling

### PR DESCRIPTION
**Pull request by CsatariGergely**
closes: #898

### CODE REVIEW

**Comment by pgoyal01** [comment]
Sorry about the Typo issue.

1.  "images with vulnerabilities" must be blocked from being loaded into the registries.
1. For the suggestion, "Modify "Image Signing" to "Image Certification (Scanning for vulnerabilities, basic operations, ..) and Signing" -- "basic operations" refers to operations that can be performed on the images in the repository. For example, if existing images cannot be modified then that is not a permitted operation and should be tested/certified.

File: doc/ref_arch/kubernetes/chapters/chapter05.md [hunk 1]
```diff
@@ -120,7 +120,7 @@ Secrets must not be stored in scripts or code but provided dynamically at runtim
 ##  5.13  Trusted Registry
 Ensure that the container registry only accepts container images from trusted sources that have tested and validated the images. Where images are provided by third parties, define and follow a formal process to validate compliance with security requirements. Also ensure that access control is applied to registries requiring unique credentials, to limit who can control the build or add images.
 
- - Network access to the registry must run over TLS or VPN connections
+- Network access to the registry must run over TLS or VPN connections
```
Comment (tomkivlin) [inline doc/ref_arch/kubernetes/chapters/chapter05.md:123]: Can the 'must' be changed to guidance / principle wording please?

File: doc/ref_arch/kubernetes/chapters/chapter05.md [hunk 2]
```diff
@@ -31,9 +31,9 @@ It is recommended that:
 - Image registries are monitored to automatically block or replace images with known vulnerabilities, while also ensuring policies are used to gate what can be deployed and who can deploy from the registry.
 - Adopt a layered packaging model which supports separation of concerns during image build.
 
-The following functionalities are recommended for securing kubernetes platform;
+The following functionalities are recommended for securing kubernetes platform:
```
Comment (xmulligan) [inline doc/ref_arch/kubernetes/chapters/chapter05.md:34]:
```suggestion
The following functionalities are recommended for securing Kubernetes platforms:
```

File: doc/ref_arch/kubernetes/chapters/chapter05.md [hunk 3]
```diff
@@ -31,9 +31,9 @@ It is recommended that:
...
 
-- Image Signing
+- Image Certification (Scan for voulnerabilities) and Signing
```
Comment (xmulligan) [inline doc/ref_arch/kubernetes/chapters/chapter05.md:36]:
```suggestion
- Image Certification (Scan for vulnerabilities) and Signing
```

File: doc/ref_arch/kubernetes/chapters/chapter05.md [hunk 4]
```diff
@@ -31,9 +31,9 @@ It is recommended that:
...
 - Role-base Access Control
 - Secret Managment
```
Comment (rihabbanday) [inline doc/ref_arch/kubernetes/chapters/chapter05.md:38]:
```suggestion
- Secret Management
```

**Comment by rabi-abdel** [comment]
@CsatariGergely can you please address comments above?

**Comment by rabi-abdel** [comment]
@CsatariGergely can you please have a look at the input above?

File: doc/ref_arch/kubernetes/chapters/chapter05.md (hunk 1 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/chapter05.md:123]: Humm, not sure. Can we trust a registry if the connection is optionally secure to it? What others think?

Comment (xmulligan) [reply doc/ref_arch/kubernetes/chapters/chapter05.md:123]: I would agree with @CsatariGergely

Comment (walterkozlowski) [reply doc/ref_arch/kubernetes/chapters/chapter05.md:123]: I would agree "in principle" with @CsatariGergely and @xmulligan. My suggestion use the following statement:
"Network access to the registry must run over a secured connections, such as TLS, SSl, or VPN."
Just to  not limit to a particular protocol.
@CsatariGergely, please address this discussion and we will accept this PR.

Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter05.md:123]: Would the following work?  I don't disagree that it's needed, it's just the language I'm wary of as this isn't part of the specification, it's guidance.
```suggestion
- It is strongly recommended that network access to the registry is secured using TLS or VPN connections to ensure trust.
```

**Comment by tomkivlin** [comment]
> I would agree "in principle" with @CsatariGergely and @xmulligan. My suggestion use the following statement:
> "Network access to the registry must run over a secured connections, such as TLS, SSl, or VPN."
> Just to not limit to a particular protocol.
> @CsatariGergely, please address this discussion and we will accept this PR.

@walterkozlowski can you review the changes @CsatariGergely made please and mark as approved if you're happy so we can merge.

**Pull request merged**